hyper/http11/connection.py

```diff
@@ -113,19 +115,24 @@ def connect(self):
         :returns: Nothing.
         """
         if self._sock is None:
-            if not self.proxy_host:
-                host = self.host
-                port = self.port
-            else:
-                host = self.proxy_host
-                port = self.proxy_port
 
-            sock = socket.create_connection((host, port), 5)
+            if self.proxy_host and self.secure:
+                # http CONNECT proxy
+                sock = HTTP11Connection._create_tunnel(
+                    self.proxy_host, self.proxy_port, self.host, self.port)
+            else:
```

Comment:
This branch is probably simpler as two `elif` branches that just consist of a one-line call to `socket.create_connection`.